### PR TITLE
Unit Tests for Movement Module

### PR DESCRIPTION
**Pull request by ctlai95**

### CODE REVIEW

File: src/movement/movement.lua [hunk 1]
```diff
@@ -54,7 +54,7 @@ end
 function Movement:moveUp()
     moveable = true
     for _, square in pairs(self.board.currentTetromino.squares) do
-        if square.y < self.board.height or self.board.boardTetrominosMatrix[square.x][square.y + 1] ~= 0 then
+        if square.y + 1 >= self.board.height or self.board.boardTetrominosMatrix[square.x][square.y + 1] ~= 0 then
```
Comment (henrytran28) [inline src/movement/movement.lua]: LOL I can't believe we missed this. Also could you change this and the other movements to not isFilled too 😸

File: src/movement/movement_test.lua [hunk 2]
```diff
@@ -0,0 +1,296 @@
+local Movement = require("movement/movement")
+local Board = require("board/board")
+local Tetromino = require("tetromino/tetromino")
+local properties = require("tetromino/properties")
+local Point = require("point/point")
+local Square = require("square/square")
+local colors = require("colors/colors")
+
+function getTetrominoWidth(tetromino)
```
Comment (henrytran28) [inline src/movement/movement_test.lua]: Can you change these 2 functions to local

File: src/movement/movement_test.lua [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    before_each(function()
+        movement = Movement:init(Board:new(10, 22))
+        ids = {"O", "I", "J", "L", "S", "Z", "T"}
+    end)
+
+    it("Constructor", function()
+        assert.is_truthy(movement.board)
+    end)
+
+    it("getTetrominoWidth", function()
```
Comment (henrytran28) [inline src/movement/movement_test.lua]: GetTetrominoWidth to be consistent right? Same with other functions

File: src/movement/movement_test.lua [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+            assert.are_same(Point:new(4, 5 - tetrominoHeight), movement.board.currentTetromino.origin)
+            movement:moveUp() -- blocked by occupied squares
+            assert.are_same(Point:new(4, 5 - tetrominoHeight), movement.board.currentTetromino.origin)
+            movement.board.boardTetrominoSquares = {}
+        end
+    end)
+
+    it("hardDrop", function()
+        for _, id in pairs(ids) do
+            -- initialize some variables
```
Comment (henrytran28) [inline src/movement/movement_test.lua]: don't need this comment lol

**Review by henrytran28 — CHANGES_REQUESTED** [review]
Partial review for now. Will review again once the other PR gets in.

File: src/movement/movement.lua (hunk 1 above)
Comment (ctlai95) [reply src/movement/movement.lua]: actually I caught this one in the python version of this PR, it just never got merged so I didn't fix it here lol
But yea good catch on the !isFilled

File: src/movement/movement_test.lua (hunk 3 above)
Comment (ctlai95) [reply src/movement/movement_test.lua]: I was gonna incorporate widths/heights at different rotation states but I didn't end up using it I think.

Comment (henrytran28) [reply src/movement/movement_test.lua]: so is this never used?

Comment (henrytran28) [reply src/movement/movement_test.lua]: oh so should these be removed?

Comment (ctlai95) [reply src/movement/movement_test.lua]: eh i think its fine to leave them

**Comment by ctlai95** [comment]
@henrytran28 ready for review :)

**Review by henrytran28 — APPROVED** [review]

**Pull request merged**